ext/mbstring/tests/utf_encodings.phpt

```diff
@@ -0,0 +1,1023 @@
+--TEST--
+Torture test for UTF-{7,8,16,32}
+--EXTENSIONS--
+mbstring
+--SKIPIF--
+<?php
+if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+?>
+--FILE--
+<?php
+srand(232); /* Make results consistent */
+mb_substitute_character(0x25); // '%'
+include('encoding_tests.inc');
+
+// all ranges of valid codepoints in UnicodeData.txt
+$validRanges = [
```

Comment:
Is it necessary to use a fine grained breakdown? I'd have expected that `[0x0, 0xd8000]` for the low codepoints would do, as UTF encodings shouldn't care which codepoints are actually allocated.

Reply:
> UTF encodings shouldn't care which codepoints are actually allocated

I guess you meant `0xd800`? Sure, I can change those lower ranges to `[0, 0xd800]` if you like. Personally I don't care much either way.

What do you say?

Reply:
> > UTF encodings shouldn't care which codepoints are actually allocated
> 
> I guess you meant `0xd800`?

Right :)

> Sure, I can change those lower ranges to `[0, 0xd800]` if you like. Personally I don't care much either way.
> 
> What do you say?

I don't particularly care either ;) The list is likely to get outdated, but it's not like it really matters.

Ideally we'd just use the whole `0-0x10ffff` range minus surrogates and not care about the details, but I expect that this would make the code much slower as it would cover about 10x more code points ^^ Dunno how slow it is right now.

Reply:
Well, if you don't care, and I don't care...